### ISSUE TITLE
Unable to connect to DC/OS Marathon provider

### ISSUE THREAD

**Issue by willgorman** [containous/traefik#1510]
### What version of Traefik are you using?
```
Version:      v1.2.3
Codename:     morbier
Go version:   go1.7.5
Built:        2017-04-13_07:32:59PM
OS/Arch:      darwin/amd64
```

### What is your environment & configuration (arguments, toml...)?
```json
{
 "GraceTimeOut": 10,
 "Debug": false,
 "CheckNewVersion": true,
 "AccessLogsFile": "",
 "TraefikLogsFile": "",
 "LogLevel": "DEBUG",
 "EntryPoints": {},
 "Cluster": null,
 "Constraints": [],
 "ACME": null,
 "DefaultEntryPoints": [],
 "ProvidersThrottleDuration": 2000000000,
 "MaxIdleConnsPerHost": 200,
 "InsecureSkipVerify": false,
 "Retry": null,
 "Docker": null,
 "File": null,
 "Web": null,
 "Marathon": {
  "Watch": true,
  "Filename": "",
  "Constraints": [],
  "Endpoint": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
  "Domain": "marathon.mesos",
  "ExposedByDefault": true,
  "GroupsAsSubDomains": false,
  "DCOSToken": "***REDACTED***",
  "MarathonLBCompatibility": true,
  "TLS": null,
  "DialerTimeout": 60,
  "KeepAlive": 10,
  "Basic": null
 },
 "Consul": null,
 "ConsulCatalog": null,
 "Etcd": null,
 "Zookeeper": null,
 "Boltdb": null,
 "Kubernetes": null,
 "Mesos": null,
 "Eureka": null,
 "ECS": null,
 "Rancher": null,
 "ConfigFile": "/Users/wg9059/Downloads/traefik.toml"
}
```

### What did you do?

Attempted to start traefik with a connection to marathon (DC/OS).   The version of the DC/OS cluster is 1.9 (enterprise)

### What did you expect to see?

The process start with no errors.

### What did you see instead?

```
ERRO[2017-04-28T16:03:22-05:00] Failed to register for events, 404: <html>
<head><title>404 Not Found</title></head>
<body bgcolor="white">
<center><h1>404 Not Found</h1></center>
<hr><center>openresty/1.9.15.1</center>
</body>
</html>
```
Observing the requests with Charles proxy I can see that it is making a call to the endpoint at `http://example.com/v2/events` but since this is a DC/OS marathon it should be using `http://example.com/marathon/v2/events`.  From what I can see in the go-marathon library it looks like it should be automatically adding the /marathon to the path whenever the DCOSToken is present in the config.

If I modify the Marathon endpoint to be http://example.com/marathon instead of just http://example.com then the  error that I get instead is `ERRO[2017-04-28T16:05:33-05:00] Failed to create a client for marathon, error: Marathon API error: URI not found: /marathon/v2/apps` and by observing the requests through Charles Proxy I can see that it connects to the events stream at `http://example.com/marathon/v2/events` but tries to get the apps from `http://example.com/marathon/marathon/v2/apps`.

**Comment by timoreimann** [comment]
Thanks for the bug report.

This _might_ already be fixed in the latest go-marathon (specifically, by gambol99/go-marathon#267). I'll take a closer look and report back.

**Comment by timoreimann** [comment]
@willgorman I haven't come around digging deeper into the yet; however, what I managed to do was to push an change of Traefik 1.2.3 with just go-marathon and the presumed fix to Docker Hub.

Do you possibly have a chance to give [timoreimann/traefik:7129c29](https://hub.docker.com/r/timoreimann/traefik/tags/) a whirl?

**Comment by willgorman** [comment]
Thanks!  It looks like that does resolve the issue and traefik is able to connect to the event stream and get the apps from Marathon.  Unfortunately, go-marathon appears to be failing to deserialize the responses, which I'm guessing is probably related to some of the schema changes mentioned on https://github.com/gambol99/go-marathon/issues/171 but that'd be a separate issue.

**Comment by timoreimann** [comment]
It's probably related to the change in environment variables, specifically secrets. gambol99/go-marathon#252 is probably best related. An in-flight PR is available but hasn't addressed review comments in a while.
Feel free to ping / carry as you see fit. :-)

In your opinion, does it make sense now to release Traefik with an updated go-marathon version? Or would that be valuable only if the deserialization problem got fixed as well?

**Comment by willgorman** [comment]
I tried again after removing any apps using secrets from the cluster and everything worked so it does look like it's related to that issue.   I think the fix from updating the go-marathon version here is probably useful even without the serialization fix, at least for DC/OS users that aren't using secrets in their apps.

**Comment by timoreimann** [comment]
@willgorman gotcha, thanks for the feedback.

By coincidence, an update of go-marathon is in order to fix a different issue (see PR 1519). I expect/hope for it to still be included for 1.3, so the upcoming release should fix the problem.

**Comment by timoreimann** [comment]
Fixed by 1524. Will be included in the upcoming 1.3.
